src/System.Drawing.Common/tests/BitmapTests.cs

```diff
@@ -0,0 +1,1674 @@
+                Assert.Equal(IntPtr.Zero, bitmapData.Scan0);
+            }
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void LockBits_Disposed_ThrowsArgumentException()
+        {
+            var bitmap = new Bitmap(1, 1);
+            bitmap.Dispose();
+            Assert.Throws<ArgumentException>(null, () => bitmap.LockBits(new Rectangle(0, 0, 1, 1), ImageLockMode.ReadOnly, PixelFormat.Format32bppArgb));
+
+            var bitmapData = new BitmapData();
+            Assert.Throws<ArgumentException>(null, () => bitmap.LockBits(new Rectangle(0, 0, 1, 1), ImageLockMode.ReadOnly, PixelFormat.Format32bppArgb, bitmapData));
+            Assert.Equal(IntPtr.Zero, bitmapData.Scan0);
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void LockBits_AlreadyLocked_ThrowsInvalidOperationException()
+        {
+            using (var bitmap = new Bitmap(1, 1))
+            {
+                bitmap.LockBits(new Rectangle(0, 0, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat);
+
+                Assert.Throws<InvalidOperationException>(() => bitmap.LockBits(new Rectangle(0, 0, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat));
+                Assert.Throws<InvalidOperationException>(() => bitmap.LockBits(new Rectangle(0, 0, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat, new BitmapData()));
+
+                Assert.Throws<InvalidOperationException>(() => bitmap.LockBits(new Rectangle(1, 1, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat));
+                Assert.Throws<InvalidOperationException>(() => bitmap.LockBits(new Rectangle(1, 1, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat, new BitmapData()));
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [InlineData(0, -1)]
+        [InlineData(0, 2)]
+        [InlineData(1, 2)]
+        public void UnlockBits_InvalidHeightWidth_Nop(int offset, int invalidParameter)
+        {
+            using (var bitmap = new Bitmap(2, 2))
+            {
+                BitmapData data = bitmap.LockBits(new Rectangle(offset, offset, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat);
+                data.Height = invalidParameter;
+                data.Width = invalidParameter;
+
+                bitmap.UnlockBits(data);
+            }
+        }
+        
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void UnlockBits_Scan0Zero_Nop()
+        {
+            using (var bitmap = new Bitmap(1, 1))
+            {
+                BitmapData data = bitmap.LockBits(new Rectangle(0, 0, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat);
+                data.Scan0 = IntPtr.Zero;
+
```

Comment:
Ah bummer, I didn't think to recompile the source before running the tests after I fixed the merge conflict